misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlLexer.java

```diff
@@ -330,9 +341,11 @@ private void processSyntaxTrivia(List<STNode> triviaList, boolean isLeading) {
                     break;
                 case LexerTerminals.CARRIAGE_RETURN:
                 case LexerTerminals.NEWLINE:
-                    triviaList.add(processEndOfLine());
-                    if (isLeading) {
-                        break;
+                    if (isInitialTrivia) {
```

Comment:
Why do we need to special case first token?

Reply:
Let's discuss here,
https://github.com/ballerina-platform/ballerina-lang/pull/26429#issuecomment-720410806